pkg/psmdb/mongos.go

```diff
@@ -0,0 +1,361 @@
+package psmdb
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	api "github.com/percona/percona-server-mongodb-operator/pkg/apis/psmdb/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func MongosDeployment(cr *api.PerconaServerMongoDB) *appsv1.Deployment {
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name + "-" + "mongos",
+			Namespace: cr.Namespace,
+		},
+	}
+}
+
+func MongosDeploymentSpec(cr *api.PerconaServerMongoDB, operatorPod corev1.Pod) (appsv1.DeploymentSpec, error) {
+	ls := map[string]string{
+		"app.kubernetes.io/component": "mongos",
+		"app.kubernetes.io/instance":  cr.Name,
+	}
+
+	c, err := mongosContainer(cr)
+	if err != nil {
+		return appsv1.DeploymentSpec{}, fmt.Errorf("failed to create container %v", err)
+	}
+
+	initContainers := initContainers(cr, operatorPod)
+	for i := range initContainers {
+		initContainers[i].Resources.Limits = c.Resources.Limits
+		initContainers[i].Resources.Requests = c.Resources.Requests
+	}
+
+	return appsv1.DeploymentSpec{
+		Replicas: &cr.Spec.Sharding.Mongos.Size,
+		Selector: &metav1.LabelSelector{
+			MatchLabels: ls,
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels:      ls,
+				Annotations: cr.Spec.Sharding.Mongos.MultiAZ.Annotations,
+			},
+			Spec: corev1.PodSpec{
+				SecurityContext:   cr.Spec.Sharding.Mongos.PodSecurityContext,
+				Affinity:          PodAffinity(cr.Spec.Sharding.Mongos.MultiAZ.Affinity, ls),
+				NodeSelector:      cr.Spec.Sharding.Mongos.MultiAZ.NodeSelector,
+				Tolerations:       cr.Spec.Sharding.Mongos.MultiAZ.Tolerations,
+				PriorityClassName: cr.Spec.Sharding.Mongos.MultiAZ.PriorityClassName,
+				RestartPolicy:     corev1.RestartPolicyAlways,
+				ImagePullSecrets:  cr.Spec.ImagePullSecrets,
+				Containers:        []corev1.Container{c},
+				InitContainers:    initContainers,
+				Volumes:           volumes(cr),
+				SchedulerName:     cr.Spec.SchedulerName,
+			},
+		},
+	}, nil
+}
+
+func initContainers(cr *api.PerconaServerMongoDB, operatorPod corev1.Pod) []corev1.Container {
+	inits := []corev1.Container{}
+	if cr.CompareVersion("1.5.0") >= 0 {
```

Comment:
why this `if` needed?

Reply:
`if` not needed https://github.com/percona/percona-server-mongodb-operator/pull/439/commits/be7bdefaa2b6f362495a8a77e1b79e5e50ad2560

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/439/commits/08792f2bbbaf8c8011cf2fcd04b1f6f16c0e33c8